NaiveBayes.py

```diff
@@ -0,0 +1,204 @@
+	def classify(self, dataSet):
+
+		dataFrame = dataSet.dataFrame
+		groundTruth = dataSet.trueLabels
+
+		#### NEW #####
+		#classificationList = dataFrame[groundTruth].unique() --->this isn't going to work for the loop anymore
+		classificationList = self.model[-1] #variable that points to the dictionary of classification probabilities
+		##############
+
+		#make a new column for the data frame where our classifications are going to go
+		classificationColumn = []
+
+		#for each of the rows (people) in the dataset
+		for row in dataFrame.iterrows():
+
+			#dictionary {key = classification, value = complete bayesian probability}
+			bayesianDict = {}
+			#dictionary {key = classification, value = numerator probability}
+			numeratorDict = {}
+			denominatorSum = 0 #reset it for every row
+
+			#iterate through the possible outcomes of the class variable
+			for classification in classificationList.keys():
+
+				numeratorDict[classification] = classificationList[classification]
+
+				#loop through outer array of the model (but we stop at second to last element of array)
+				for j, attributeDict in enumerate(self.model):
+
+					#skip the last element because this isn't an attribute -- it's the classification probabilities dictionary
+					if(j == len(self.model) - 1):
+						continue
+					#if we run into the blank ground truth column, skip this
+					if(dataSet.headers[j] == dataSet.trueLabels):
+						continue
+
+					#value for the current row of the given attribute
+					attrValue = row[1].iloc[j]
+
+					if(dataSet.headers[j] in dataSet.getNumericalColumns()): #numerical
+						meanDict = attributeDict["mean"]
+						stdDict = attributeDict["std"]
+
+						#NUMERATOR = P(person|classification) * P(classification)
+						#### New classify() additions below --->  ####
+						#Now instead of calling the attributeCategoryProbability() function we're just accessing the classification value from the model
+
+						bayesNumerator = self.calculateGaussianProbability(meanDict[classification], stdDict[classification], row[1].iloc[j])
+						numeratorDict[classification] *= bayesNumerator
+					else:
+						bayesNumerator = attributeDict[attrValue][classification]
+						numeratorDict[classification] *= bayesNumerator
+
+			for key in numeratorDict.keys():
+				denominatorSum += numeratorDict[key]
+			#currently just adding dictionary of all probabilities given all classifications but eventually want to be adding the max of these (the final classification)
+			for key in numeratorDict.keys():
+				bayesianDict[key] = round(numeratorDict[key] / denominatorSum, 2)
+
+			maxClassification = max(bayesianDict.items(), key=operator.itemgetter(1))[0]
+			classificationColumn.append(maxClassification)
+		
+		#sets new column equal to the array of classifications
+
+		dataFrame["Bayes Classification"] = classificationColumn
+		print(dataFrame.to_string())
+		#dataFrame.to_csv('out.csv', sep='\t', encoding='utf-8')
+		return dataFrame
```

Comment:
Does this need to return or should it just update the DataSet object/Bayes object?